tools/redhat/redhat_osv/csaf.py

```diff
@@ -2,9 +2,59 @@
 import json
 from dataclasses import dataclass, InitVar, field
 from typing import Any, Iterable
+from urllib.parse import urlparse, parse_qs, unquote
+
 
 class RemediationParseError(ValueError):
-    pass
+    """Exception raised when parsing remediation data fails."""
+
+
+def parse_purl_info(purl: str) -> tuple[str, str]:
+    """
+    Parse a PURL to extract component name and version.
+
+    Args:
+        purl: A PURL string like "pkg:rpm/redhat/package@version?arch=x86_64"
+
+    Returns:
+        tuple: (component_name, version) where version includes epoch if present
+        Note: The version is URL-decoded (e.g., %2B becomes +)
+
+    Raises:
+        ValueError: If the PURL format is invalid or not an RPM PURL
+    """
+    if not purl.startswith("pkg:rpm/"):
+        raise ValueError(f"Only RPM PURLs are supported, got: {purl}")
+
+    # Parse the PURL: pkg:rpm/namespace/name@version?qualifiers
+    parsed = urlparse(purl)
```

Comment:
Perhaps using [packageurl-python](https://github.com/package-url/packageurl-python) instead of urllib & manual splitting might be more reliable

Reply:
Ok, I'm using packageurl-python now.